### PR TITLE
Changed ATTs infrastructure to newest ConfigureEndpointTestExecution

### PR DESCRIPTION
**Pull request by danielmarbach**
Connects to https://github.com/Particular/NServiceBus/pull/3514#issuecomment-189050351

### CODE REVIEW

**Comment by SeanFeldman** [comment]
@danielmarbach this is failing. I'll have a loot at it tomorrow. Anything I should know what changed? (204 files is a lot :) )

**Comment by danielmarbach** [comment]
Essentially only the implementation of `IConfigureEndpointTestExecution` and the `ISupportScenario...`

File: src/AcceptanceTests/ConfigureEndpointAzureServiceBusTransport.cs [hunk 1]
```diff
@@ -17,13 +17,16 @@ public class ConfigureAzureServiceBusTransport : IConfigureTestExecution
         typeof(AllNativeMultiQueueTransactionTransports)
         //typeof(AllNativeMultiQueueTransactionTransports)
```
Comment (SeanFeldman) [inline src/AcceptanceTests/ConfigureEndpointAzureServiceBusTransport.cs]: Suspect this should have been `AllTransportsWithoutNativeDeferralAndWithAtomicSendAndReceive`, which causes `When_timeout_dispatch_fails_on_timeout_data_removal` to fail.

File: src/AcceptanceTests/App_Packages/NSB.AcceptanceTests.6.0.0-unstable1607/Routing/When_publishing_to_scaled_out_subscribers_on_multicast_transports.cs [hunk 2]
```diff
@@ -3,13 +3,13 @@
...
     using NServiceBus.AcceptanceTesting;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
     using NServiceBus.AcceptanceTests.ScenarioDescriptors;
     using NUnit.Framework;
 
     public class When_publishing_to_scaled_out_subscribers_on_multicast_transports : NServiceBusAcceptanceTest
     {
-        // TODO: address when core is update
-        [Test, Explicit("Requires a fix in core before can be executed")]
+        [Test]
```
Comment (SeanFeldman) [inline src/AcceptanceTests/App_Packages/NSB.AcceptanceTests.6.0.0-unstable1607/Routing/When_publishing_to_scaled_out_subscribers_on_multicast_transports.cs]: Not sure why this got removed. Until instance-id issue is resolved with core, this test won't run.

**Pull request merged**